rust/src/push/base_rules.rs

```diff
@@ -71,7 +71,7 @@ pub const BASE_APPEND_OVERRIDE_RULES: &[PushRule] = &[
         priority_class: 5,
         conditions: Cow::Borrowed(&[Condition::Known(KnownCondition::EventMatch(
             EventMatchCondition {
-                key: Cow::Borrowed("content.m.relates_to.rel_type"),
+                key: Cow::Borrowed("content.m\\.relates_to.rel_type"),
```

Comment:
I think this is the only reasonable thing to do here -- make it abide by the new behavior?

Reply:
It does make it hard to have this behind a configuration variable though? I'm not really sure how to handle this.

Thoughts are welcome!

Reply:
Is this change visible to users? I suppose we could represent keys unambiguously inside Synapse and degrade them to ambiguous form when matching / presenting them to users and the experimental flag is disabled.

Then that raises the question of how custom push rules would work when a server admin toggles the experimental flag. We could also record whether a key is in msc3873 format, but that would add a bunch of complexity to a fairly simple PR.

Reply:
> Is this change visible to users?

No, this rule is disabled by default. Mostly Synapse needs to be internally consistent here?

> I suppose we could represent keys unambiguously inside Synapse and degrade them to ambiguous form when matching / presenting them to users and the experimental flag is disabled.

The MSC originally had this and we removed it.